From f3398f8e48203246d6346b5c7c4ec5ea065181e6 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Fri, 19 Jul 2019 15:01:49 +0100
Subject: [PATCH] Do not trigger an inotity event when the AppStream XML data
 is unchanged

We already call dnf_utils_refresh_repo_appstream() on startup, but only as
required. Remove the unconditional copy to speed up gnome-software startup.
---
 backends/dnf/pk-backend-dnf.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/backends/dnf/pk-backend-dnf.c b/backends/dnf/pk-backend-dnf.c
index 779896c2d..503a6eb0a 100644
--- a/backends/dnf/pk-backend-dnf.c
+++ b/backends/dnf/pk-backend-dnf.c
@@ -565,13 +565,6 @@ dnf_utils_add_remote (PkBackendJob *job,
 	if (!ret)
 		return FALSE;
 
-	/* update the AppStream copies in /var */
-	for (guint i = 0; i < repos->len; i++) {
-		DnfRepo *repo = g_ptr_array_index (repos, i);
-		if (!dnf_utils_refresh_repo_appstream (repo, error))
-			return FALSE;
-	}
-
 	/* done */
 	if (!dnf_state_done (state, error))
 		return FALSE;
-- 
2.21.0

